cmd/juju/caas/add_test.go

```diff
@@ -280,6 +329,152 @@ func (s *addCAASSuite) TestNonExistClusterName(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, `clusterName \"non existing cluster name\" not found`)
 }
 
+type regionTestCase struct {
+	title          string
+	regionStr      string
+	expectedErrStr string
+}
+
+func (s *addCAASSuite) TestRegionFlag(c *gc.C) {
+	for _, ts := range []regionTestCase{
+		{
+			title:          "missing cloud",
+			regionStr:      "/region",
+			expectedErrStr: `validating cloud region "/region": parsing cloud region: expected <cloudType>/<region> not valid`,
```

Comment:
This error message is poorly worded, hence the request t change the code to construct the NotValid error. see how " not valid" is appended? errors.NotValidf("foo") results in "foo not valid". If you want a NotValid error with a raw string, you use errors.NewNotValid(nil, "foo")